qtp_target_gene/fastqc.py

```diff
@@ -0,0 +1,271 @@
+    """Generates the FastQC commands
+
+    Parameters
+    ----------
+    seqs : list of str
+        The list of seqs filepaths
+    map_file : str
+        The path to the mapping file
+    out_dir : str
+        The job output directory
+
+    Returns
+    -------
+    list of str
+        The FastQC commands
+
+    Raises
+    ------
+
+    Notes
+    -----
+    This is presently reproducing functionality from the kneaddata pipeline
+    in qp_shotgun.
+    """
+
+    samples = make_read_pairs_per_sample(forward_seqs, reverse_seqs, map_file)
+
+    cmds = []
+    prefixes = []
+
+    fps = []
+    for run_prefix, sample, f_fp, r_fp in samples:
+        prefixes.append(run_prefix)
+        if r_fp is None:
+            cmds.append('mkdir -p %s; fastqc --outdir "%s" --noextract %s' %
+                        (join(out_dir, run_prefix), join(out_dir, run_prefix),
+                         f_fp))
+            fps.append((f_fp, None))
+        else:
+            cmds.append('mkdir -p %s; fastqc --outdir "%s" --noextract %s %s' %
+                        (join(out_dir, run_prefix), join(out_dir, run_prefix),
+                         f_fp, r_fp))
+            fps.append((f_fp, r_fp))
+
+    return cmds, samples
+
+
+def generate_multiqc_commands(file_dir, out_dir):
+    """Generates the multiqc commands
+
+    Parameters
+    ----------
+    file_dir : str
+        Filepath to directory of QC info
+    out_dir : str
+        The job output directory
+
+    Returns
+    -------
+    list of str
+        The multiqc commands
+
+    Notes
+    -----
+    """
+    # make input file list
+    cmds = []
+
+    cmds.append('multiqc --outdir %s --filename MultiQC.html %s'
+                '%s' % (out_dir, file_dir))
+
+    return cmds
+
+
+def _run_commands(qclient, job_id, commands, msg):
+    for i, cmd in enumerate(commands):
+        qclient.update_job_step(job_id, msg % i)
+        std_out, std_err, return_value = system_call(cmd)
+        if return_value != 0:
+            error_msg = ("Error running FastQC:\nStd out: %s\nStd err: %s"
+                         "\n\nCommand run was:\n%s"
+                         % (std_out, std_err, cmd))
+            return False, error_msg
+
+    return True, ""
+
+
+def fastqc(qclient, job_id, map_file, filepaths):
+    """Run FastQC and MultiQC on a sequence artifact
+
+    Parameters
+    ----------
+    qclient : qiita_client.QiitaClient
+        The Qiita server client
+    filepaths: list of tup
+        List of per_sample_fastq filepath artifacts from validate function
+    map_file : str
+        The path to the mapping file
+    Returns
+    -------
+    bool, list of tup, str
+        The results of the job
+    """
+
+    # Get forward and (if exists) reverse seq filepaths
+    fwd_fps = [x for x, y in filepaths if y == 'raw_forward_seqs']
+    rev_fps = [x for x, y in filepaths if y == 'raw_reverse_seqs']
+
+    # Generate temp dir for exection of QC steps
+    out_dir = mkdtemp()
```

Comment:
@josenavas Currently this is creating a temporary directory for output, which worked fine when this was operating as a plugin. How do we do this as part of validate.py? I see that generate_html_summary takes out_dir as a required argument -- should we do the same here, and if so, how do we determine it?